smartsim/_core/generation/generator.py

```diff
@@ -131,24 +132,26 @@ def _gen_exp_dir(self) -> None:
         else:
             logger.info("Working in previously created experiment")
 
-    def _gen_orc_dir(self, orchestrator: t.Optional[Orchestrator]) -> None:
+    def _gen_orc_dir(self, orchestrator_list: t.Optional[t.List[Orchestrator]]) -> None:
+        # orchestrator: t.Optional[Orchestrator]
         """Create the directory that will hold the error, output and
            configuration files for the orchestrator.
 
         :param orchestrator: Orchestrator instance
         :type orchestrator: Orchestrator | None
         """
-
-        if not orchestrator:
+        # No orchestrators
+        if not orchestrator_list:
```

Comment:
If my above comment makes sense, then `orchestrator_list` is guaranteed to be a list, no need to check if it is falsy -> an empty list will not be iterated over.